lib/src/Uncertainty/Distribution/LeastSquaresFactory.cxx

```diff
@@ -0,0 +1,414 @@
+String LeastSquaresFactory::__repr__() const
+{
+  OSS oss(true);
+  oss << "class=" << this->getClassName()
+      << " distribution=" << distribution_
+      << " solver=" << solver_;
+  return oss;
+}
+
+/* String converter */
+String LeastSquaresFactory::__str__(const String & ) const
+{
+  return this->getClassName();
+}
+
+class LeastSquaresFactoryResidualEvaluation : public EvaluationImplementation
+{
+public:
+  LeastSquaresFactoryResidualEvaluation(const Sample & sample,
+                          const Distribution & distribution,
+                          const Point & knownParameterValues,
+                          const Indices & knownParameterIndices)
+    : EvaluationImplementation()
+    , sample_(sample)
+    , distribution_(distribution)
+    , knownParameterValues_(knownParameterValues)
+    , knownParameterIndices_(knownParameterIndices)
+  {
+    // build the unknown indices
+    const UnsignedInteger effectiveParameterSize = distribution.getParameter().getSize();
+    for (UnsignedInteger j = 0; j < effectiveParameterSize; ++ j)
+    {
+      if (!knownParameterIndices_.contains(j))
+        unknownParameterIndices_.add(j);
+    }
+    empiricalCDF_ = Point(sample.getSize());
+    for(UnsignedInteger i = 0 ; i < sample.getSize(); ++ i)
+      empiricalCDF_[i] = sample.computeEmpiricalCDF(sample[i]);
+  }
+
+  LeastSquaresFactoryResidualEvaluation * clone() const
+  {
+    return new LeastSquaresFactoryResidualEvaluation(*this);
+  }
+
+  UnsignedInteger getInputDimension() const
+  {
+    return unknownParameterIndices_.getSize();
+  }
+
+  UnsignedInteger getOutputDimension() const
+  {
+    return sample_.getSize();
+  }
+
+  Description getInputDescription() const
+  {
+    return Description::BuildDefault(getInputDimension(), "theta");
+  }
+
+  Description getOutputDescription() const
+  {
+    return Description(sample_.getSize(), "r");
+  }
+
+  Description getDescription() const
+  {
+    Description description(getInputDescription());
+    description.add(getOutputDescription());
+    return description;
+  }
+
+  Point operator() (const Point & parameter) const
+  {
+    // Define conditinned distribution
+    Distribution distribution(distribution_);
+    Point effectiveParameter(distribution.getParameter());
+    // set unknown values
+    UnsignedInteger unknownParameterSize = unknownParameterIndices_.getSize();
+    for (UnsignedInteger j = 0; j < unknownParameterSize; ++ j)
+    {
+      effectiveParameter[unknownParameterIndices_[j]] = parameter[j];
+    }
+    // set known values
+    UnsignedInteger knownParametersSize = knownParameterIndices_.getSize();
+    for (UnsignedInteger j = 0; j < knownParametersSize; ++ j)
+    {
+      effectiveParameter[knownParameterIndices_[j]] = knownParameterValues_[j];
+    }
+    try
+    {
+      distribution.setParameter(effectiveParameter);
+    }
+    catch (Exception &)
+    {
+      return Point(getOutputDimension(), SpecFunc::LogMinScalar);
```

Comment:
why 'Min'?
If we want to minimize residuals, maybe we should set value to 'Max'